domain_models/views.py

```diff
@@ -0,0 +1,170 @@
+"""Contextual view module."""
+from . import models
+import six
+
+
+class ContextViewMetaClass(type):
+    """Context view meta class."""
+
+    attributes = dict()
+
+    def __new__(mcs, class_name, bases, attributes):
+        """Context view class factory."""
+        mcs.attributes = attributes
```

Comment:
Not sure that it's a good idea to store `attributes` in class variable. Perhaps, it would be better to provide them as validate method's argument:

```python
mcs.validate(attributes, bases)
```